lib/utils/utils.go

```diff
@@ -269,6 +269,21 @@ func SliceContainsStr(slice []string, value string) bool {
 	return false
 }
 
+// SliceRemove makes a copy of the slice and removes the passed in values from the copy.
+func SliceRemove(slice []string, values ...string) []string {
+	// Always operate on a copy to be on the safe side.
+	slice = CopyStrings(slice)
```

Comment:
why not make a slice with allocated cap but empty and add to it instead of removing?